components/bt/host/bluedroid/stack/btm/btm_inq.c

```diff
@@ -2537,7 +2537,69 @@ void BTM_AddEirService( UINT32 *p_eir_uuid, UINT16 uuid16 )
 
 /*******************************************************************************
 **
-** Function         BTM_RemoveEirService
+** Function         btm_find_custom_uuid_slots
+**
+** Description      Helper function for custom service managing routines.
+**
+** Parameters       custom_uuid - pointer to custom_uuid array in tBTA_DM_CB
+**                  uuid - UUID struct
+**
+** Returns          None
+**
+*******************************************************************************/
+static void btm_find_custom_uuid_slots(tBT_UUID *custom_uuid, tBT_UUID uuid,
+                                         UINT8 *empty_slot, UINT8 *match_slot)
+{
+    *empty_slot = BTA_EIR_SERVER_NUM_CUSTOM_UUID;
+    *match_slot = BTA_EIR_SERVER_NUM_CUSTOM_UUID;
+
+    for ( UINT8 xx = 0; xx < BTA_EIR_SERVER_NUM_CUSTOM_UUID; xx++ ) {
+        if ( custom_uuid[xx].len == 0 ) {
+            if ( *empty_slot == BTA_EIR_SERVER_NUM_CUSTOM_UUID )
+                *empty_slot = xx;
+        } else if ( *match_slot == BTA_EIR_SERVER_NUM_CUSTOM_UUID ) {
+            if ( !memcmp (&custom_uuid[xx].uu, &uuid.uu, uuid.len) ) {
+                *match_slot = xx;
+            }
+        }
+    }
+}
+
+/*******************************************************************************
+**
+** Function         BTM_AddCustomEirService
+**
+** Description      This function is called to add a custom UUID.
+**
+** Parameters       custom_uuid - pointer to custom_uuid array in tBTA_DM_CB
+**                  uuid - UUID struct
+**
+** Returns          None
+**
+*******************************************************************************/
+void BTM_AddCustomEirService( tBT_UUID *custom_uuid, tBT_UUID uuid )
+{
+    UINT8 empty_slot, match_slot;
+
+    btm_find_custom_uuid_slots(custom_uuid, uuid, &empty_slot, &match_slot);
+
+    if (match_slot == BTA_EIR_SERVER_NUM_CUSTOM_UUID) {
+        if (empty_slot == BTA_EIR_SERVER_NUM_CUSTOM_UUID) {
+            BTM_TRACE_ERROR("No space to add UUID for EIR");
+            return;
+        } else {
+            memcpy (&(custom_uuid[empty_slot]), &(uuid), sizeof(tBT_UUID));
+            BTM_TRACE_EVENT("UUID saved in %d slot\r\n", empty_slot);
+        }
+    } else {
+        BTM_TRACE_WARNING("UUID is already added for EIR");
```

Comment:
If the UUID has been added to the EIR, there should be no need to call `bta_dm_set_eir` in `bta_dm_eir_update_uuid`.

Reply:
I added the `BTM_HasCustomEirService()` function to avoid this situation.